### ISSUE TITLE
Changelog is not being written to repository.

### ISSUE THREAD

**Issue by BeyondEvil** [semantic-release/changelog#65]
Here's some selected output from `--debug` (entire output at the bottom):

```
[9:23:04 PM] [semantic-release] › ✔  Run automated release from branch develop
  semantic-release:git Error: Command failed with exit code 128: git push --dry-run git@github.com:proxyco/testing-semantic-release.git HEAD:develop
  semantic-release:git     at makeError (/usr/local/lib/node_modules/semantic-release/node_modules/execa/lib/error.js:56:11)
  semantic-release:git     at handlePromise (/usr/local/lib/node_modules/semantic-release/node_modules/execa/index.js:114:26)
  semantic-release:git     at process._tickCallback (internal/process/next_tick.js:68:7) +0ms
[9:23:04 PM] [semantic-release] › ✔  Allowed to push to the Git repository

*snip*

[9:23:05 PM] [semantic-release] › ℹ  Start step "prepare" of plugin "@semantic-release/changelog"
[9:23:05 PM] [semantic-release] [@semantic-release/changelog] › ℹ  Create /root/project/CHANGELOG.stage.md
[9:23:05 PM] [semantic-release] › ✔  Completed step "prepare" of plugin "@semantic-release/changelog"
[9:23:05 PM] [semantic-release] › ℹ  Start step "prepare" of plugin "@semantic-release/git"
  semantic-release:git globed assets: [] +0ms
[9:23:05 PM] [semantic-release] [@semantic-release/git] › ℹ  Prepared Git release: stage-v1.0.4
[9:23:05 PM] [semantic-release] › ✔  Completed step "prepare" of plugin "@semantic-release/git"
[9:23:07 PM] [semantic-release] › ✔  Created tag stage-v1.0.4
```

Config:
```
module.exports = {
    "branch": "develop",
    "preset": "eslint",
    "plugins": [
        "@semantic-release/commit-analyzer",
        [
            "@semantic-release/release-notes-generator",
            {
                "linkCompare": false,
                "writerOpts": {
                    "headerPartial": "{{#if isPatch}}##{{else}}#{{/if}} {{#if @root.linkCompare}}[Stage]({{@root.host}}/{{#if @root.owner}}{{@root.owner}}/{{/if}}{{@root.repository}}/compare/{{previousTag}}...{{currentTag}}){{else}}Stage{{/if}}{{#if title}} \"{{title}}\"{{/if}}{{#if date}} ({{date}}){{/if}}",
                    // This overrides: https://github.com/conventional-changelog/conventional-changelog/blob/master/packages/conventional-changelog-eslint/writer-opts.js#L24
                    "transform": (commit) => {
                        if (!commit.tag || typeof commit.tag !== `string`) { 
                            return 
                        }
                        commit.message = commit.message.replace(/NNN-\d+/g, '[$&](https://some.link.com/$&)');
                        commit.shortHash = commit.hash.substring(0, 7); 
                        return commit
                    }
                }
            }
        ],
        [
            "@semantic-release/changelog",
            {
                "changelogFile": "CHANGELOG.stage.md"
            }
        ],
        [
            "@semantic-release/git",
            {
                "message": "${nextRelease.gitTag} released. [skip ci]\n\n${nextRelease.notes}"
            }
        ],
        [
            "@semantic-release/github",
            {
                "successComment": false
            }
        ],
        [
            "semantic-release-slack-bot",
            {
                "notifyOnSuccess": true,
                "markdownReleaseNotes": true,
                "onSuccessTemplate": {
                    "text": "Updates to $package_name has been released to *Stage*!\n\n$release_notes"
                }
            }
        ]
    ]
}
```

I don't understand why it doesn't get written. What is that error?

<details>

<summary> entire log </summary>

[9:23:04 PM] [semantic-release] › ℹ  Running semantic-release version 15.13.28
  semantic-release:config load config from: /root/project/.releaserc.js +0ms
  semantic-release:config options values: { branch: 'develop',
  semantic-release:config   repositoryUrl: 'git@github.com:proxyco/testing-semantic-release.git',
  semantic-release:config   tagFormat: 'stage-v${version}',
  semantic-release:config   plugins:
  semantic-release:config    [ '@semantic-release/commit-analyzer',
  semantic-release:config      [ '@semantic-release/release-notes-generator', [Object] ],
  semantic-release:config      [ '@semantic-release/changelog', [Object] ],
  semantic-release:config      [ '@semantic-release/git', [Object] ],
  semantic-release:config      [ '@semantic-release/github', [Object] ],
  semantic-release:config      [ 'semantic-release-slack-bot', [Object] ] ],
  semantic-release:config   preset: 'eslint',
  semantic-release:config   _: [],
  semantic-release:config   'tag-format': 'stage-v${version}',
  semantic-release:config   t: 'stage-v${version}',
  semantic-release:config   debug: true,
  semantic-release:config   '$0': 'semantic-release' } +6ms
  semantic-release:plugins options for @semantic-release/changelog/verifyConditions: { changelogFile: 'CHANGELOG.stage.md' } +0ms
[9:23:04 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/changelog"
  semantic-release:plugins options for @semantic-release/git/verifyConditions: { message:
  semantic-release:plugins    '${nextRelease.gitTag} released. [skip ci]\n\n${nextRelease.notes}' } +1ms
[9:23:04 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/git"
  semantic-release:plugins options for @semantic-release/github/verifyConditions: { successComment: false } +0ms
[9:23:04 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/github"
  semantic-release:plugins options for semantic-release-slack-bot/verifyConditions: { notifyOnSuccess: true,
  semantic-release:plugins   markdownReleaseNotes: true,
  semantic-release:plugins   onSuccessTemplate:
  semantic-release:plugins    { text:
  semantic-release:plugins       'Updates to $package_name has been released to *Stage*!\n\n$release_notes' } } +0ms
[9:23:04 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "semantic-release-slack-bot"
  semantic-release:plugins options for @semantic-release/commit-analyzer/analyzeCommits: {} +1ms
[9:23:04 PM] [semantic-release] › ✔  Loaded plugin "analyzeCommits" from "@semantic-release/commit-analyzer"
  semantic-release:plugins options for @semantic-release/release-notes-generator/generateNotes: { linkCompare: false,
  semantic-release:plugins   writerOpts:
  semantic-release:plugins    { headerPartial:
  semantic-release:plugins       '{{#if isPatch}}##{{else}}#{{/if}} {{#if @root.linkCompare}}[Stage]({{@root.host}}/{{#if @root.owner}}{{@root.owner}}/{{/if}}{{@root.repository}}/compare/{{previousTag}}...{{currentTag}}){{else}}Stage{{/if}}{{#if title}} "{{title}}"{{/if}}{{#if date}} ({{date}}){{/if}}',
  semantic-release:plugins      transform: [Function: transform] } } +0ms
[9:23:04 PM] [semantic-release] › ✔  Loaded plugin "generateNotes" from "@semantic-release/release-notes-generator"
  semantic-release:plugins options for @semantic-release/changelog/prepare: { changelogFile: 'CHANGELOG.stage.md' } +0ms
[9:23:04 PM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/changelog"
  semantic-release:plugins options for @semantic-release/git/prepare: { message:
  semantic-release:plugins    '${nextRelease.gitTag} released. [skip ci]\n\n${nextRelease.notes}' } +0ms
[9:23:04 PM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/git"
  semantic-release:plugins options for @semantic-release/github/publish: { successComment: false } +1ms
[9:23:04 PM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/github"
  semantic-release:plugins options for @semantic-release/github/success: { successComment: false } +0ms
[9:23:04 PM] [semantic-release] › ✔  Loaded plugin "success" from "@semantic-release/github"
  semantic-release:plugins options for semantic-release-slack-bot/success: { notifyOnSuccess: true,
  semantic-release:plugins   markdownReleaseNotes: true,
  semantic-release:plugins   onSuccessTemplate:
  semantic-release:plugins    { text:
  semantic-release:plugins       'Updates to $package_name has been released to *Stage*!\n\n$release_notes' } } +0ms
[9:23:04 PM] [semantic-release] › ✔  Loaded plugin "success" from "semantic-release-slack-bot"
  semantic-release:plugins options for @semantic-release/github/fail: { successComment: false } +0ms
[9:23:04 PM] [semantic-release] › ✔  Loaded plugin "fail" from "@semantic-release/github"
  semantic-release:plugins options for semantic-release-slack-bot/fail: { notifyOnSuccess: true,
  semantic-release:plugins   markdownReleaseNotes: true,
  semantic-release:plugins   onSuccessTemplate:
  semantic-release:plugins    { text:
  semantic-release:plugins       'Updates to $package_name has been released to *Stage*!\n\n$release_notes' } } +1ms
[9:23:04 PM] [semantic-release] › ✔  Loaded plugin "fail" from "semantic-release-slack-bot"
[9:23:04 PM] [semantic-release] › ✔  Run automated release from branch develop
  semantic-release:git Error: Command failed with exit code 128: git push --dry-run git@github.com:proxyco/testing-semantic-release.git HEAD:develop
  semantic-release:git     at makeError (/usr/local/lib/node_modules/semantic-release/node_modules/execa/lib/error.js:56:11)
  semantic-release:git     at handlePromise (/usr/local/lib/node_modules/semantic-release/node_modules/execa/index.js:114:26)
  semantic-release:git     at process._tickCallback (internal/process/next_tick.js:68:7) +0ms
[9:23:04 PM] [semantic-release] › ✔  Allowed to push to the Git repository
[9:23:04 PM] [semantic-release] › ℹ  Start step "verifyConditions" of plugin "@semantic-release/changelog"
[9:23:04 PM] [semantic-release] › ✔  Completed step "verifyConditions" of plugin "@semantic-release/changelog"
[9:23:04 PM] [semantic-release] › ℹ  Start step "verifyConditions" of plugin "@semantic-release/git"
[9:23:04 PM] [semantic-release] › ✔  Completed step "verifyConditions" of plugin "@semantic-release/git"
[9:23:04 PM] [semantic-release] › ℹ  Start step "verifyConditions" of plugin "@semantic-release/github"
[9:23:04 PM] [semantic-release] [@semantic-release/github] › ℹ  Verify GitHub authentication
[9:23:04 PM] [semantic-release] › ✔  Completed step "verifyConditions" of plugin "@semantic-release/github"
[9:23:04 PM] [semantic-release] › ℹ  Start step "verifyConditions" of plugin "semantic-release-slack-bot"
[9:23:04 PM] [semantic-release] › ✔  Completed step "verifyConditions" of plugin "semantic-release-slack-bot"
  semantic-release:get-last-release found tags: [ { gitTag: 'stage-v1.0.3', version: '1.0.3' }, { gitTag: 'stage-v1.0.2', version: '1.0.2' }, { gitTag: 'stage-v1.0.1', version: '1.0.1' }, { gitTag: 'stage-v1.0.0', version: '1.0.0' } ] +0ms
[9:23:05 PM] [semantic-release] › ℹ  Found git tag stage-v1.0.3 associated with version 1.0.3
  semantic-release:get-commits Use gitHead: 50e1de9fec86160de74463d4b50faf8e7af4b006 +0ms
[9:23:05 PM] [semantic-release] › ℹ  Found 3 commits since last release
  semantic-release:get-commits Parsed commits: [ { commit: { long: '8d784e0b0be351b24fe1ea4e4f13f61d54cd8a20', short: '8d784e0' }, tree: { long: 'a1a40d586cae63f3a94ef70f642d262e5a7fba31', short: 'a1a40d5' }, author: { name: 'Jim Brännlund', email: 'jimbrannlund@fastmail.com', date: 2019-10-31T21:18:45.000Z }, committer: { name: 'Jim Brännlund', email: 'jimbrannlund@fastmail.com', date: 2019-10-31T21:18:45.000Z }, subject: 'Fix: Remove all test-files', body: '', hash: '8d784e0b0be351b24fe1ea4e4f13f61d54cd8a20', message: 'Fix: Remove all test-files', gitTags: '(HEAD, origin/develop, develop)', committerDate: 2019-10-31T21:18:45.000Z }, { commit: { long: 'ba62e06b147f40fae61e56de1b701bfed259b431', short: 'ba62e06' }, tree: { long: 'fc68c8dfdc2652714f7be6a2e342f56a08127acc', short: 'fc68c8d' }, author: { name: 'Jim Brännlund', email: 'BeyondEvil@users.noreply.github.com', date: 2019-10-31T21:12:47.000Z }, committer: { name: 'GitHub', email: 'noreply@github.com', date: 2019-10-31T21:12:47.000Z }, subject: 'Merge pull request #39 from proxyco/debug-changelog', body: 'Chore: debug changelog in semantic-release', hash: 'ba62e06b147f40fae61e56de1b701bfed259b431', message: 'Merge pull request #39 from proxyco/debug-changelog\n\nChore: debug changelog in semantic-release', gitTags: '', committerDate: 2019-10-31T21:12:47.000Z }, { commit: { long: '1e9a90510f318d5dd4c9c8cce0324b77dd19b29c', short: '1e9a905' }, tree: { long: 'fc68c8dfdc2652714f7be6a2e342f56a08127acc', short: 'fc68c8d' }, author: { name: 'Jim Brännlund', email: 'jimbrannlund@fastmail.com', date: 2019-10-31T21:11:40.000Z }, committer: { name: 'Jim Brännlund', email: 'jimbrannlund@fastmail.com', date: 2019-10-31T21:11:40.000Z }, subject: 'Chore: debug changelog in semantic-release', body: '', hash: '1e9a90510f318d5dd4c9c8cce0324b77dd19b29c', message: 'Chore: debug changelog in semantic-release', gitTags: '', committerDate: 2019-10-31T21:11:40.000Z } ] +10ms
[9:23:05 PM] [semantic-release] › ℹ  Start step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[9:23:05 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: Fix: Remove all test-files
  semantic-release:commit-analyzer Analyzing with default rules +0ms
  semantic-release:commit-analyzer The rule { tag: 'Fix', release: 'patch' } match commit with release type 'patch' +0ms
[9:23:05 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is patch
[9:23:05 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: Merge pull request #39 from proxyco/debug-changelog

Chore: debug changelog in semantic-release
  semantic-release:commit-analyzer Analyzing with default rules +1ms
[9:23:05 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[9:23:05 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: Chore: debug changelog in semantic-release
  semantic-release:commit-analyzer Analyzing with default rules +0ms
[9:23:05 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[9:23:05 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analysis of 3 commits complete: patch release
[9:23:05 PM] [semantic-release] › ✔  Completed step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[9:23:05 PM] [semantic-release] › ℹ  The next release version is 1.0.4
[9:23:05 PM] [semantic-release] › ℹ  Start step "generateNotes" of plugin "@semantic-release/release-notes-generator"
  semantic-release:release-notes-generator version: '1.0.4' +0ms
  semantic-release:release-notes-generator host: undefined +0ms
  semantic-release:release-notes-generator owner: 'proxyco' +1ms
  semantic-release:release-notes-generator repository: 'testing-semantic-release' +0ms
  semantic-release:release-notes-generator previousTag: 'stage-v1.0.3' +0ms
  semantic-release:release-notes-generator currentTag: 'stage-v1.0.4' +0ms
  semantic-release:release-notes-generator host: 'https://github.com' +0ms
  semantic-release:release-notes-generator host: 'https://github.com' +0ms
  semantic-release:release-notes-generator linkReferences: undefined +0ms
  semantic-release:release-notes-generator issue: 'issues' +0ms
  semantic-release:release-notes-generator commit: 'commit' +0ms
[9:23:05 PM] [semantic-release] › ✔  Completed step "generateNotes" of plugin "@semantic-release/release-notes-generator"
[9:23:05 PM] [semantic-release] › ℹ  Start step "prepare" of plugin "@semantic-release/changelog"
[9:23:05 PM] [semantic-release] [@semantic-release/changelog] › ℹ  Create /root/project/CHANGELOG.stage.md
[9:23:05 PM] [semantic-release] › ✔  Completed step "prepare" of plugin "@semantic-release/changelog"
[9:23:05 PM] [semantic-release] › ℹ  Start step "prepare" of plugin "@semantic-release/git"
  semantic-release:git globed assets: [] +0ms
[9:23:05 PM] [semantic-release] [@semantic-release/git] › ℹ  Prepared Git release: stage-v1.0.4
[9:23:05 PM] [semantic-release] › ✔  Completed step "prepare" of plugin "@semantic-release/git"
[9:23:07 PM] [semantic-release] › ✔  Created tag stage-v1.0.4
[9:23:07 PM] [semantic-release] › ℹ  Start step "publish" of plugin "@semantic-release/github"
  semantic-release:github release owner: 'proxyco' +0ms
  semantic-release:github release repo: 'testing-semantic-release' +1ms
  semantic-release:github release name: 'stage-v1.0.4' +0ms
  semantic-release:github release branch: 'develop' +0ms
[9:23:08 PM] [semantic-release] [@semantic-release/github] › ℹ  Published GitHub release: https://github.com/proxyco/testing-semantic-release/releases/tag/stage-v1.0.4
[9:23:08 PM] [semantic-release] › ✔  Completed step "publish" of plugin "@semantic-release/github"
[9:23:08 PM] [semantic-release] › ℹ  Start step "success" of plugin "@semantic-release/github"
[9:23:08 PM] [semantic-release] [@semantic-release/github] › ℹ  Skip commenting on issues and pull requests.
  semantic-release:github found semantic-release issues: [] +0ms
[9:23:08 PM] [semantic-release] › ✔  Completed step "success" of plugin "@semantic-release/github"
[9:23:08 PM] [semantic-release] › ℹ  Start step "success" of plugin "semantic-release-slack-bot"
[9:23:08 PM] [semantic-release] [semantic-release-slack-bot] › ℹ  Sending slack notification on success
[9:23:08 PM] [semantic-release] [semantic-release-slack-bot] › ℹ  { text:
   'Updates to Testing Semantic Release has been released to *Stage*!\n\n*Stage (2019-10-31)*\n\n*Chore*\n\n•   debug changelog in semantic-release (<https://github.com/proxyco/testing-semantic-release/commit/1e9a90510f318d5dd4c9c8cce0324b77dd19b29c|1e9a905>)\n\n*Fix*\n\n•   Remove all test-files (<https://github.com/proxyco/testing-semantic-release/commit/8d784e0b0be351b24fe1ea4e4f13f61d54cd8a20|8d784e0>)\n' }
[9:23:08 PM] [semantic-release] › ✔  Completed step "success" of plugin "semantic-release-slack-bot"
[9:23:08 PM] [semantic-release] › ✔  Published release 1.0.4

</details>

**Comment by BeyondEvil** [comment]
When running against `master`, the `CHANGELOG.md` gets written. So I don't think the issue is the `GH_TOKEN`. (`develop` is the default branch in the repo).

Config:
```
module.exports = {
    "branch": "master",
    "preset": "eslint",
    "plugins": [
        "@semantic-release/commit-analyzer",
        [
            "@semantic-release/release-notes-generator",
            {
                "writerOpts": {
                    "transform": (commit) => {
                        if (!commit.tag || typeof commit.tag !== `string`) { 
                            return 
                        }
                        commit.message = commit.message.replace(/NNN-\d+/g, '[$&](https://some.link.com/$&)');
                        commit.shortHash = commit.hash.substring(0, 7); 
                        return commit
                    }
                }
            }
        ],
        "@semantic-release/changelog",
        "@semantic-release/git",
        "@semantic-release/github",
        [
            "semantic-release-slack-bot",
            {
                "notifyOnSuccess": true,
                "markdownReleaseNotes": true
            }
        ]
    ]
}
```

**Comment by BeyondEvil** [comment]
I found a difference between the outputs of `master` and `develop`:

master:
```
[9:27:00 PM] [semantic-release] › ℹ  Start step "prepare" of plugin "@semantic-release/git"
  semantic-release:git globed assets: [ 'CHANGELOG.md' ] +0ms
```

develop:
```
[9:23:05 PM] [semantic-release] › ℹ  Start step "prepare" of plugin "@semantic-release/git"
  semantic-release:git globed assets: [] +0ms
```

I can't find anything in the docs for `changelog` that says that you have to configure `git` as well. Even the example doesn't do that, where the changelog is moved to `docs/CHANGELOG.md`.

**Comment by pvdlg** [comment]
The file is committed and pushed to the remote repo by `@semantic-release/git`. As [indicated in the documentation](https://github.com/semantic-release/git#options) by default `@semantic-release/git` commit `['CHANGELOG.md', 'package.json', 'package-lock.json', 'npm-shrinkwrap.json']`.

If you write the change in `docs/CHANGELOG.md` then you have to configure the `assets` option of `@semantic-release/git` to match that.

If you use the default option for `@semantic-release/changelog` (`./CHANGELOG.md`) then it will works because it match the defaults of `@semantic-release/git`.

I fixed the example in #66.

**Comment by tebaly** [comment]
This is **BUG** in version 8 and 9
> Command failed: git commit CHANGELOG.md -m chore(release): ...

I downgrade to 4.2.* - and **error gone**
